### PR TITLE
Create artefact store

### PR DESCRIPTION
**Pull request by hiker**
The artefact store is originally just a dict. This introduces a new object that as the artefact store (which inherits from dict, so it behaves the same), but it allows us to add additional methods if required.

It also adds a getter in the BuildConfig (to avoid accessing the private member `self._artefact_store`)). I also fixed a few coding style issues (again something we need to figure out :) )

### CODE REVIEW

**Comment by hiker** [comment]
Taking this back, it actually fails some CI tests that I must have missed :(

File: source/fab/artefacts.py
Comment (jasonjunweilyu) [inline source/fab/artefacts.py:1]: Line 164: `is indexed by a string`.

**Review by jasonjunweilyu — APPROVED** [review]
Hi @hiker , I have reviewed the changes related to this pull request. They all look good to me with only one typo found. As the tests have all passed, I have approved this.

File: source/fab/artefacts.py
Comment (hiker) [reply source/fab/artefacts.py:1]: Fixed.

**Comment by hiker** [comment]
Thanks for the review. I've updated the PR, flake8 and CI seems to be fine. Ready for next review,

**Review by jasonjunweilyu — APPROVED** [review]
Thanks. Now I think it is ready to merge.

**Pull request merged**